pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -182,6 +198,15 @@ type Template struct {
 	// Name is the name of the template
 	Name string `json:"name"`
 
+	// Template is the name of the template which is used as the base of this template.
+	Template string `json:"template,omitempty"`
+
+	// Arguments hold arguments to the template.
+	Arguments Arguments `json:"arguments,omitempty"`
+
+	// TemplateRef is the reference to the template resource which is used as the base of this template.
+	TemplateRef *TemplateRef `json:"templateRef,omitempty"`
```

Comment:
Are you sure this belongs to the `Template` struct? AFAIK it should be in the `DagTask` struct alongside the `Template` field.